plugins/org.wso2.developerstudio.eclipse.platform.core/src/org/wso2/developerstudio/eclipse/platform/core/startup/alert/JavaVersionAlertHandler.java

```diff
@@ -62,23 +63,24 @@ public void run() {
 
 						shell.setLocation(xCenter, yCenter);
 
-						String warningMessage =
-						                        Constants.DETECTED_JAVA_VERSION_MESSAGE + version + "\n" +
-						                                Constants.RECOMMENDED_JAVA_VERSION_MESSAGE +
-						                                Constants.MINIMUM_REQUIRED_JAVA_VERSION + ".x" + "\n" +
-						                                Constants.JAVA_VERSION_ALERT_MESSAGE + "\n" +
-						                                Constants.JAVA_VERSION_CONTINUE_MESSAGE;
-						MessageDialog dialog =
-						                       new MessageDialog(shell, ALERT_TITLE, null, warningMessage,
-						                               MessageDialog.ERROR,
-						                               new String[] { CONTINUE_BUTTON, EXIT_BUTTON }, 0) {
-							                       protected void buttonPressed(int buttonId) {
-								                       if (buttonId == 1) {
-									                       System.exit(0);
-								                       }
-								                       super.buttonPressed(buttonId);
-							                       }
-						                       };
+						String str = (Constants.MAXIMUM_SUPPORTED_JAVA_VERSION
+								== Constants.MINIMUM_REQUIRED_JAVA_VERSION) ?
+								"\n" :
+								(" - " + Constants.MAXIMUM_SUPPORTED_JAVA_VERSION + ".x" + "\n");
+
+						String warningMessage = Constants.DETECTED_JAVA_VERSION_MESSAGE + version + "\n"
+								+ Constants.RECOMMENDED_JAVA_VERSION_MESSAGE + Constants.MINIMUM_REQUIRED_JAVA_VERSION
+								+ ".x" + str + Constants.JAVA_VERSION_ALERT_MESSAGE;
+
+						MessageDialog dialog = new MessageDialog(shell, ALERT_TITLE, null, warningMessage,
+								MessageDialog.ERROR, new String[] { EXIT_BUTTON }, 0) {
+							protected void buttonPressed(int buttonId) {
+								if (buttonId == 0) {
+									System.exit(0);
+								}
+								super.buttonPressed(buttonId);
```

Comment:
Will this be invoked since we have only one button?

Reply:
removed as it is not and we are going to exit anyway.